### DISCUSSION TITLE
$PSVersionTable mit Out-GridView und Out-ConsoleGridView unterschiedlich angezeigt?

### DISCUSSION THREAD

**Discussion by helmutkilian** [TobiasPSP/OReilly#8] [Fragen] [answered]
Ich verstehe nicht, warum das Feld $PSVersionTable.PSCompatibleVersions
mit 
$PSVersionTable | Out-ConsoleGridView 
vollständig angezeigt wird, also alle Versionsnummern tatsächlich zu sehen sind (ohne geschweifte Klammern), und mit 
$PSVersionTable | Out-GridView
ist nur {1.0 2.0 3.0 4.0 ...} zu sehen!?

**Comment by TobiasPSP** [comment, marked as answer]
`Out-GridView` enthält Logik, die nur die "wichtigsten" Eigenschaften (Spalten) anzeigt, so ähnlich wie wenn man Informationen direkt in die PowerShell-Konsole ausgibt. Dazu gehört auch die "Verkürzung" von Arrays. Der Inhalt von PSCompatibleVersions ist solch ein Array, und per Default werden nur die ersten vier Elemente angezeigt. Möchte man lieber ALLE Array-Elemente anzeigen, setzt man die Variable `$FormatEnumerationLimit` auf den Wert -4 (der Normalwert beträgt 4).

**Reply by helmutkilian** [reply]
Vielen Dank für ihre schnelle Antwort (und auch für die Antworten zu zwei weiteren Beiträgen von mir)!

`$FormatEnumerationLimit` habe ich ausprobiert. Tatsächlich liefert der Wert 0 nur die geschweiften Klammern mit drei Punkten darin.
Für ALLE Elemente meinten Sie vermutlich `$FormatEnumerationLimit < 0`, denn schon bei -1 sehe ich in dem von mir genannten Beispiel  alle Elemente.

**Reply by TobiasPSP** [reply]
Absolut richtig, -1 ist der Wert, um alles zu sehen. Das passiert offenbar, wenn man ohne Powershell und mit Computerverbot belegt mit dem Smartphone aus dem Urlaub kommentiert 😊 ab kommende Woche bin ich wieder vollausgestattet im Büro.
